src/function/gds/page_rank.cpp

```diff
@@ -35,35 +37,209 @@ struct PageRankBindData final : public GDSBindData {
     }
 };
 
-class PageRankOutputWriter {
+static void addCAS(std::atomic<double>& origin, double valToAdd) {
+    auto expected = origin.load(std::memory_order_relaxed);
+    auto desired = expected + valToAdd;
+    while (!origin.compare_exchange_strong(expected, desired))
```

Comment:
This is confusing to me. Don't you need to update `desired` inside the while loop? I think this is a bug.

Reply:
Done